warehouse/integrations/deltalake/deltalake.go

```diff
@@ -896,7 +896,7 @@ func (d *Deltalake) hasAWSCredentials() bool {
 
 // partitionQuery returns a query to fetch partitions for a table
 func (d *Deltalake) partitionQuery(ctx context.Context, tableName string) (string, error) {
-	if !d.config.enablePartitionPruning {
+	if !d.config.enablePartitionPruning || d.Uploader.ShouldOnDedupUseNewRecord() {
```

Comment:
Since the changes are straightforward, going forward with the changes. Also, I have captured the test [here](https://github.com/rudderlabs/rudder-server/pull/3851/files#diff-8ad3c6da0af87741a94ff8d4986b4fb69b117a490a3c8c705e15c97681e43d99R555-R596) since it was easy for me to add it there.